test/e2e/operator_test.go

```diff
@@ -192,24 +192,86 @@ func TestDefaultIngressClass(t *testing.T) {
 	// 	expected          = "true"
 	// )
 	// if actual, ok := ingressclass.Annotations[defaultAnnotation]; !ok {
-	// 	t.Fatalf("ingressclass %q has no %q annotation", name, defaultAnnotation)
+	// 	t.Fatalf("ingressclass %q has no %q annotation", name.Name, defaultAnnotation)
 	// } else if actual != expected {
 	// 	t.Fatalf("expected %q annotation to have value %q, found %q", defaultAnnotation, expected, actual)
 	// }
 
 	// The controller should recreate the ingressclass if it is deleted.
 	if err := kclient.Delete(context.TODO(), ingressclass); err != nil {
-		t.Fatalf("failed to delete ingressclass %q: %v", name, err)
+		t.Fatalf("failed to delete ingressclass %q: %v", name.Name, err)
 	}
 	err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
 		if err := kclient.Get(context.TODO(), name, ingressclass); err != nil {
-			t.Logf("failed to get ingressclass %q: %v", name, err)
+			t.Logf("failed to get ingressclass %q: %v", name.Name, err)
 			return false, nil
 		}
 		return true, nil
 	})
 	if err != nil {
-		t.Fatalf("failed to observe recreated ingressclass %q: %v", name, err)
+		t.Fatalf("failed to observe recreated ingressclass %q: %v", name.Name, err)
+	}
+}
+
+// TestCustomIngressClass verifies that the ingressclass controller creates an
+// ingressclass for a custom ingresscontroller and deletes the ingressclass if
+// the ingresscontroller is deleted.
+func TestCustomIngressClass(t *testing.T) {
```

Comment:
Does the operator have deletion privileges for ingress classes?

Reply:
Sigh.  Fixed.